dynamiq/nodes/tools/stagehand_tool.py

```diff
@@ -0,0 +1,189 @@
+- Extract structured information using natural language prompts.
+- Perform actions like clicks, selections, and form filling.
+
+### Parameters
+- `action_type`: Must be one of the following:
+  - `goto`: Navigate to the specified URL.
+  - `observe`: Return a list of candidate DOM elements based on the instruction.
+  - `extract`: Extract structured data as described in the instruction.
+  - `act`: Perform a user-specified action (e.g., click a button, select an option).
+- `instruction`: A natural language prompt specifying the action to perform. Required for all actions except `goto`.
+- `url`: The web address to navigate to. Required only when `action_type` is `goto`.
+
+### Usage Examples
+1. Navigate to a web page:
+   ```json
+   {
+     "action_type": "goto",
+     "url": "https://example.com"
+   }
+    ````
+
+2. Observe candidate elements:
+
+   ```json
+   {
+     "action_type": "observe",
+     "instruction": "Find all clickable links on the homepage"
+   }
+   ```
+
+3. Extract data:
+
+   ```json
+   {
+     "action_type": "extract",
+     "instruction": "Get the list of product names and prices"
+   }
+   ```
+
+4. Perform an action:
+
+   ```json
+   {
+     "action_type": "act",
+     "instruction": "Click the 'Add to Cart' button for the first product"
+   }
+   ```
+
+### Tips
+
+- Use clear, specific instructions for better extraction and action reliability.
+- Break down complex tasks into sequential steps.
+- `observe` is useful for debugging or when planning a follow-up action.
+"""
+
+
+class StagehandActionType(str, Enum):
+    ACT = "act"
+    EXTRACT = "extract"
+    OBSERVE = "observe"
+    GOTO = "goto"
+
+
+class StagehandInputSchema(BaseModel):
+    action_type: StagehandActionType = Field(
+        ...,
+        description="""
+    Specifies the type of interaction with the web page:
+
+    - `act`: Perform an action on the current page (e.g., click, fill, select).
+    - `extract`: Retrieve structured data from the page based on the given instruction.
+    - `observe`: Return a list of candidate DOM elements for potential interaction.
+    - `goto`: Navigate to the specified URL.
+    """,
+    )
+
+    instruction: str | None = Field(
+        None,
+        description="""
+    A natural language instruction describing the action, data to extract, or elements to observe.
+    Required for all action types except `goto`.
+    """,
+    )
+
+    url: str | None = Field(
+        None,
+        description="""
+    The target URL to navigate to. Required only when `action_type` is `goto`.
+    """,
+    )
+
+
+class StagehandTool(ConnectionNode):
+    group: Literal[NodeGroup.TOOLS] = NodeGroup.TOOLS
+    name: str = "Stagehand Tool"
+    description: str = DESCRIPTION_STAGEHAND
+    connection: StagehandConnection
+    model_name: str
+
+    input_schema: ClassVar[type[StagehandInputSchema]] = StagehandInputSchema
+    session: Stagehand | None = None
+
+    def start_session(self):
```

Comment:
It seems instead of `start_session` and `session` param need to override `init_components` method to additionally init client like:
```
    def init_components(self, connection_manager: ConnectionManager | None = None):
        """
        Initialize components for the node.

        Args:
            connection_manager (ConnectionManager, optional): The connection manager. Defaults to ConnectionManager.
        """
        connection_manager = connection_manager or ConnectionManager()
        super().init_components(connection_manager)
        self.client.model_name = self.model_name
        self.client.init()
```
After just use `self.client`